### PR TITLE
f-data-networkfirewall_firewall_resource_policy

### PR DESCRIPTION
**Pull request by ddericco**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates #18026

Output from acceptance testing:

```
$ make testacc TESTS=TestAccNetworkFirewallFirewallResourcePolicyDataSource PKG=networkfirewall 
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/networkfirewall/... -v -count 1 -parallel 20 -run='TestAccNetworkFirewallFirewallResourcePolicyDataSource'  -timeout 180m
=== RUN   TestAccNetworkFirewallFirewallResourcePolicyDataSource
=== PAUSE TestAccNetworkFirewallFirewallResourcePolicyDataSource
=== CONT  TestAccNetworkFirewallFirewallResourcePolicyDataSource
--- PASS: TestAccNetworkFirewallFirewallResourcePolicyDataSource (141.53s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/networkfirewall     144.125s

...
```

### CODE REVIEW

File: internal/service/networkfirewall/firewall_resource_policy_data_source.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+		ReadContext: dataSourceFirewallResourcePolicyRead,
+		Schema: map[string]*schema.Schema{
+			"resource_arn": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: verify.ValidARN,
+			},
+			"policy": {
+				// No validation needed since policy JSON is returned from call
```
Comment (silvaalbert) [inline internal/service/networkfirewall/firewall_resource_policy_data_source.go:24]: Consider removing this comment, as it is implicit when setting `Computed: true` 
```suggestion

```

File: internal/service/networkfirewall/firewall_resource_policy_data_source_test.go [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+	"regexp"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/service/networkfirewall"
+	sdkacctest "github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-provider-aws/internal/acctest"
+)
+
+func TestAccNetworkFirewallFirewallResourcePolicyDataSource(t *testing.T) {
```
Comment (silvaalbert) [inline internal/service/networkfirewall/firewall_resource_policy_data_source_test.go:14]: Please rename to meet naming standards
```suggestion
func TestAccNetworkFirewallFirewallResourcePolicyDataSource_basic(t *testing.T) {
```

**Review by silvaalbert — COMMENTED** [review]
Looking good! Providing one minor suggestion regarding a comment, and a request to fix the test function name.

**Comment by ewbankkit** [comment]
@ddericco Thanks for the contribution :tada: :clap:.

**Pull request merged**